src/python.cc

```diff
@@ -2121,6 +2325,461 @@ ModelState::ModelState(TRITONBACKEND_Model* triton_model)
+    ModelConfig().Write(&buffer);
+
+    std::string model_config = buffer.MutableContents();
+
+    std::unique_ptr<IPCMessage> auto_complete_message =
+        IPCMessage::Create(shm_pool_, false /* inline_response */);
+    auto_complete_message->Command() = PYTHONSTUB_AutoCompleteRequest;
+
+    std::unique_ptr<PbString> pb_string =
+        PbString::Create(shm_pool_, model_config);
+    bi::managed_external_buffer::handle_t string_handle =
+        pb_string->ShmHandle();
+
+    auto_complete_message->Args() = string_handle;
+    stub_message_queue_->Push(auto_complete_message->ShmHandle());
+
+    std::unique_ptr<IPCMessage> auto_complete_response_message =
+        IPCMessage::LoadFromSharedMemory(
+            shm_pool_, parent_message_queue_->Pop());
+
+    if (auto_complete_response_message->Command() !=
+        PYTHONSTUB_AutoCompleteResponse) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL,
+          (std::string(
+               "Received unexpected resposne from Python backend stub: ") +
+           name_)
+              .c_str());
+    }
+
+    auto auto_complete_response =
+        std::move((shm_pool_->Load<AutoCompleteResponseShm>(
+                      auto_complete_response_message->Args())))
+            .data_;
+
+    if (auto_complete_response->response_has_error) {
+      if (auto_complete_response->response_is_error_set) {
+        std::unique_ptr<PbString> error_message =
+            PbString::LoadFromSharedMemory(
+                shm_pool_, auto_complete_response->response_error);
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INTERNAL, error_message->String().c_str());
+      } else {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INTERNAL,
+            (std::string("Auto-complete failed for ") + Name()).c_str());
+      }
+    }
+
+    if (auto_complete_response->response_has_model_config) {
+      std::unique_ptr<PbString> auto_complete_config =
+          PbString::LoadFromSharedMemory(
+              shm_pool_, auto_complete_response->response_model_config);
+      std::string auto_complete_config_string = auto_complete_config->String();
+      // Need to corret the string for the right json format
+      CorrectStringFormat(&auto_complete_config_string);
+      RETURN_IF_ERROR(auto_complete_config_.Parse(auto_complete_config_string));
+      triton::common::TritonJson::WriteBuffer buf;
+      auto_complete_config_.Write(&buf);
+    }
+
+    initialized_ = true;
+  }
+
+  return nullptr;  // success
+}
+
+void
+ModelState::CorrectStringFormat(std::string* str)
+{
+  *str = std::regex_replace(*str, std::regex("\'"), "\"");
+  *str = std::regex_replace(*str, std::regex("False"), "false");
+  *str = std::regex_replace(*str, std::regex("True"), "true");
+}
+
+TRITONSERVER_Error*
+ModelState::AutoCompleteConfig()
+{
+  RETURN_IF_ERROR(CheckModelConfigMismatch());
+
+  // If the model configuration already specifies inputs and outputs
```

Comment:
This is not accurate. We need to auto-complete missing information even if the model config already specify input and output.  The model_configuration may have specified only one model input but the model might be expecting 2.
I think you copied this logic from onnx backend. 
See the the enhancement request here: https://github.com/triton-inference-server/onnxruntime_backend/issues/113